pkg/daemon/manager/manager.go

```diff
@@ -98,6 +115,26 @@ func (m *manager) EnsureServiceState(service winsvc.Service, state svc.State) er
 			return err
 		}
 	case svc.Stopped:
+		// Before we can stop this service, we need to make sure all services dependent on this service are stopped
+		// The service must be cast to the actual type so we can get its handle
+		winSvc, ok := service.(*mgr.Service)
+		if !ok {
+			return fmt.Errorf("service is not correct type")
+		}
+		dependentServices, err := m.getDependentServices(winSvc.Handle)
+		if err != nil {
+			return errors.Wrap(err, "error finding dependent services")
+		}
+		for _, dependentServiceName := range dependentServices {
```

Comment:
What is preventing us from just using the service object as is? This would majorly simplify things

``` suggestion
		config, err := service.Config()
		if err != nil {
			return errors.Wrap(err, "error finding dependent services")
		}
		for _, dependentServiceName := range config.Dependencies {
```

Reply:
What you've suggested is looking at the services this service is dependent on, not the services that are dependent on this services.

Handling the case of kubelet, my code will stop hybrid-overlay, kube-proxy before stopping kubelet, the suggested code will stop containerd which isnt necessary.

Reply:
Ah I see, thought I was missing something. An alternative to the `EnumDependentServicesW` approach would be listing all Windows services and filtering them based on if our `service` appears in their `Dependencies` list. Simpler code but definitely more syscalls.
```
svcConfig, err := service.Config()
if err != nil {
	return nil
}
allSvcs, err := m.GetServices()
if err != nil {
	return nil
}
dependentServices := []string{}
for potentialDependent := range allSvcs {
	potentialDependentHandle, err := m.OpenService(potentialDependent)
	if err != nil {
		return nil
	}
	potentialDependentConfig, err := potentialDependentHandle.Config()
	if err != nil {
		return nil
	}
	for _, dep := range potentialDependentConfig.Dependencies {
		if dep == svcConfig.DisplayName {
			dependentServices = append(dependentServices, potentialDependent)
		}
	}
}
```